src/main/java/com/bean00/RequestProcessor.java

```diff
@@ -25,20 +26,15 @@ public Response processRequest(Request request) throws IOException {
             response = new Response(Status.OK);
         } else if (!fileCanBeFound) {
             response = new Response(Status.NOT_FOUND);
+        } else if (isHEADRequest) {
+            response = new Response(Status.OK, body, false);
```

Comment:
I think it's better not to pass in the body & not to have that boolean flag

Reply:
I'm passing in the request method now, and using that to set the `displayBody` boolean.

Reply:
can't you just not display the body if no body is passed in?

Reply:
oh, you're passing it in so that you can create the content length header out of it. Hmm.. I'm wondering if you're just passing in the body, you should be doing the same for headers. Doesn't it make more sense for `Response` to just have these things passed in & build out the response vs. having it be more of a response builder just for one part of it (headers) and not for another part (body)?

Reply:
So, later on I actually do that. I build up the headers in my `RequestProcessor`, then pass them in to the `Response` class (I'm doing this in the Media Types story)